### ISSUE TITLE
Ban/ignore null literals for non-nullable types

### ISSUE THREAD

**Issue by puzpuzpuz** [questdb/questdb#1701]
### Describe the bug

ATM the following query
```sql
select * from t where abyte = null;
```
executes just fine and returns rows with `abyte` column equal to zero. The same happens for short columns.

Boolean, byte, short and char column types don't support nulls (important note: geobyte and geoshort types support nulls), so we need to do something with this behavior. We could either report an error for queries involving byte/short null comparisons, or always evaluate such checks to `false` (even if it's something like `abyte <> null`).

### Environment

```markdown
- **QuestDB version**:
- **OS**:
- **Browser**:
```

**Comment by XinyiQiao** [comment]
What's the status of this issue? I'd like to work on this if it's still open. Thanks!

**Comment by puzpuzpuz** [comment]
> What's the status of this issue? I'd like to work on this if it's still open. Thanks!

Hi @XinyiQiao,

I don't think someone is working on this issue at the moment, so would be great if you picked it. Please check the comments on #1730 for some hints.

**Comment by XinyiQiao** [comment]
Great, I'm on it

**Comment by marregui** [comment]
hi @XinyiQiao how are you?
I am working on #1967 , I was tempted to fix this issue on non nullable types in there by always returning:
- true: for `col IS NOT NULL` when col's type is [char, byte, bool]
- false: for `col IS NULL` when col's type is [char, byte, bool]
then I saw your comment, are you working on it? if so thank you for contributing!!, do you need any help?

**Comment by XinyiQiao** [comment]
Hi @marregui Yes I'm working on it and will fix it following your logic above. I will reach out if anything comes up. Please let me know if there's any timeline I need to be aware of. Thank you!

**Comment by marregui** [comment]
Hi @XinyiQiao , that is fantastic!! Thank you again.

My own goal is to merge my branch today/tomorrow. There is no real timeline except, the sooner it gets merged, the earlier it gets out in a release, and the quicker you get it into a production environment. I find it thrilling.

**Comment by rajdeepdas2000** [comment]
I would like to work on this issue.

**Comment by joanneong** [comment]
@rajdeepdas2000 Looks like this has been fixed in merged PR #2000

**Comment by puzpuzpuz** [comment]
@joanneong thanks for the heads-up! Closing this one.
